ocean_lib/assets/test/test_asset_downloader.py

```diff
@@ -38,14 +38,15 @@ def test_ocean_assets_download_indexes(publisher_ocean_instance, metadata):
     """Tests different values of indexes that raise AssertionError."""
     publisher = get_publisher_wallet()
     metadata_copy = metadata.copy()
-    data_provider = DataServiceProvider()
+    data_provider = DataServiceProvider
 
     ddo = publisher_ocean_instance.assets.create(metadata_copy, publisher)
     wait_for_ddo(publisher_ocean_instance, ddo.did)
     sa = ServiceAgreement.from_ddo(ServiceTypes.ASSET_ACCESS, ddo)
 
     index = range(3)
-    with pytest.raises(AssertionError):
+    # AssertionError when TYPECHECK = false, else TypeError
+    with pytest.raises((AssertionError, TypeError)):
```

Comment:
can you be more explicit here? I'd recomment splitting this into more tests:
when TYPECHECK is false, assert that AssertionError is raised and TypeError is NOT raised
when TYPECHECK is true, assert that TypeError is raised and AssertionError is NOT raised

Reply:
I made this check more explicit by splitting it into two. I specify the exact error that should occur but I *don't* specify what shouldn't happen. I think this is sufficient based on https://docs.pytest.org/en/latest/reference/reference.html?highlight=xfail#pytest-raises:
>If the code block does not raise the expected exception (ZeroDivisionError in the example above), or no exception at all, the check will fail instead.